prototype2/server/src/engine/mod.rs

```diff
@@ -1,120 +1,57 @@
-mod event_handler;
-mod physics;
+mod neo_physics;
+mod neo_network;
+mod neo_player;
+mod neo_debug;
 
-use self::event_handler::EventHandler;
+use time;
 
-use std::mem;
-use std::collections::HashMap;
+use world::ServerWorld2;
 
-use time::{self, Duration};
-use itertools::Itertools;
-use uuid::Uuid;
+use specs;
 
-use common::protocol::{
-  ClientEvent,
-  ClientPayload,
-  ServerPayload,
-};
-use common::world::ClientWorld;
-use protocol::OutboundEvent;
-use world::ServerWorld;
-use world::views::player::PlayerView;
-use world::views::client_world::ClientWorldView;
-use network::Fragmentable;
-use engine::physics::{Physics, SimplePhysics};
+const NETWORK_IO_PRIORITY: specs::Priority = 100;
+const NETWORK_EVENT_DISTRIBUTION_PRIORITY: specs::Priority = 80;
+const NETWORK_HEALTH_CHECK_PRIORITY: specs::Priority = 70;
+const PHYSICS_PRIORITY: specs::Priority = 9;
+const PLAYER_CONNECTION_PRIORITY: specs::Priority = 8;
+const PLAYER_SNAPSHOT_PRIORITY: specs::Priority = 6;
+const PLAYER_INPUT_PRIORITY: specs::Priority = 5;
+const DEBUG_PRIORITY: specs::Priority = 1;
 
-pub struct ClientSnapshotHistory {
-  pub last_ack: Option<u16>,
-  pub past_snapshots: HashMap<u16, ClientWorld>
+pub struct Engine {
+  pub planner: specs::Planner<Delta>,
 }
 
-pub struct Engine {
-  // TODO: this is pub so Physics can manipulate it
-  //   add an analogue to View so it doesn't need to be pub
-  pub world: ServerWorld,
-  events: Vec<ClientPayload>,
-  physics: SimplePhysics,
-  snapshot_idx: u16,
-  client_snapshot_histories: HashMap<Uuid, ClientSnapshotHistory>
+#[derive(Debug, Clone)]
+pub struct Delta {
+  pub dt: time::Duration,
+  pub now: time::Tm,
 }
 
 impl Engine {
-  pub fn push_event(&mut self, event: ClientPayload) { self.events.push(event) }
-
-  pub fn new() -> Engine {
-    Engine {
-      world: ServerWorld::new(),
-      physics: Physics::simple_physics(),
-      events: Vec::new(),
-      snapshot_idx: 0,
-      client_snapshot_histories: HashMap::new(),
-    }
-  }
-
-  pub fn tick(&mut self, dt: &time::Duration) -> Vec<ServerPayload> {
-    let mut event_buf = Vec::new();
-    // Yank all the events off the queue, replacing with a new queue
-    mem::swap(&mut self.events, &mut event_buf);
-
-    let mut outbound: Vec<OutboundEvent> = Vec::new();
-
+  //pub fn push_event(&mut self, event: ClientPayload) { self.events.push(event) }
 
-    event_buf.drain(..).foreach(|event| outbound.append(&mut self.handle(event)));
+  pub fn new(port: u16) -> Engine {
 
-    self.physics.tick(&mut self.world, dt);
+    let neo_world = ServerWorld2::new();
+    let mut planner = specs::Planner::new(neo_world.world, 2 /* Threads, arbitrary */);
 
-    self.validate_connections();
+    planner.add_system(neo_network::io::System::new(port), "network::io", NETWORK_IO_PRIORITY);
```

Comment:
Ah this neo shit was from when i was refactoring it. Didn't want to confuse the old and the new stuff. will change